### PR TITLE
Added API schema, model, and routing for Officer Manger

### PR DESCRIPTION
**Pull request by ThaiQ**
Adding API schema, model, and routing for Officer Manager. 
With test-cases

### CODE REVIEW

File: api/models/PrintingForm3D.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ const PrintingForm3DSchema = new Schema(
 
     // unique id for query
     // id = email of the user who requested
-    id: {
+    email: {
```
Comment (evanugarte) [inline api/models/PrintingForm3D.js]: please rebase your branch. these changes were already merged in

Comment (evanugarte) [reply api/models/PrintingForm3D.js]: @ThaiQ this branch is behind. please rebase it
image

File: api/models/OfficerManager.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      type: String,
+      required: true
+    },
+    email: {
+      type: String,
+      required: true
+    },
+    facebook: {
+      type: String,
+      required: false
```
Comment (evanugarte) [inline api/models/OfficerManager.js]: the `required` field is by default false, remove if it isn't true

File: test/officerManager.js [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+let serverInstance = null
+let app = null
+const expect = chai.expect
+// tools for testing
+const tools = require('../util/testing-utils/tools.js')
+
+chai.should()
+chai.use(chaiHttp)
+
+function initializeServer () {
```
Comment (evanugarte) [inline test/officerManager.js]: In a seperate PR, we should throw `initializeServer` and `terminateServer` this into tools. I will open an issue for it.

File: test/officerManager.js [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+    tools.emptySchema(User)
+    done()
+  })
+  after(done => {
+    terminateServer(done)
+  })
+
+  let token = ''
+
+  describe('/POST submit', () => {
```
Comment (evanugarte) [inline test/officerManager.js]: for every API that requires a token, please also test that by submitting a form without a token will return `UNAUTHORIZED`.

File: test/officerManager.js [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+    })
+
+    it('Should not return statusCode 200 when the required fields are not set', done => {
+      const form = {}
+      chai
+        .request(app)
+        .post('/api/officerManager/submit')
+        .send(form)
+        .then(function (res) {
+          expect(res).to.not.have.status(200)
```
Comment (evanugarte) [inline test/officerManager.js]: change the field to be `BAD_REQUEST`. MongoDB APIs can also return 500 in some situations and it is best to test the defined behavior.

File: test/officerManager.js [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+
+          done()
+        })
+        .catch(err => {
+          throw err
+        })
+    })
+  })
+
+  describe('/POST GetForm', () => {
```
Comment (evanugarte) [inline test/officerManager.js]: for every API that requires a token, please also test that by submitting a form without a token will return `UNAUTHORIZED`.

File: test/officerManager.js [hunk 7]
```diff
@@ -0,0 +1,336 @@
...
+      chai
+        .request(app)
+        .post('/api/officerManager/GetForm')
+        .send(form)
+        .then(function (res) {
+          expect(res).to.have.status(200)
+          res.body.should.be.a('array')
+          // make sure all child has querries parameter
+          res.body.forEach(obj => {
+            if (obj.email !== 'test@test.com') {
```
Comment (evanugarte) [inline test/officerManager.js]: replace the body of `forEach` with 
```js
expect(obj.email).to.be.eql('test@test.com')
```

File: test/officerManager.js [hunk 8]
```diff
@@ -0,0 +1,336 @@
...
+          expect(res).to.have.status(404)
+
+          done()
+        })
+        .catch(err => {
+          throw err
+        })
+    })
+
+    it('Should return statusCode 200 and a message if a form was edited', done => {
```
Comment (evanugarte) [inline test/officerManager.js]: After updating, please verify that the email with this value can now be found. This is done after editing and deleting in the Event API, found [here](https://github.com/SCE-Development/Core-v4/blob/7eb49330115efe0619c97377f7b69382cca2d41d/test/Event.js#L214)

File: test/officerManager.js [hunk 9]
```diff
@@ -0,0 +1,336 @@
...
+          expect(res).to.have.status(404)
+
+          done()
+        })
+        .catch(err => {
+          throw err
+        })
+    })
+
+    it('Should return statusCode 200 and a message if a form was deleted', done => {
```
Comment (evanugarte) [inline test/officerManager.js]: After deleting, please verify that the database should now be empty. This is done after deleting in the Event API, found [here](https://github.com/SCE-Development/Core-v4/blob/7eb49330115efe0619c97377f7b69382cca2d41d/test/Event.js#L284)

File: test/officerManager.js (hunk 3 above)
Comment (evanugarte) [reply test/officerManager.js]: See #150

Comment (ThaiQ) [reply test/officerManager.js]: can you work on that. I tried to pull them over to util but having issues with the variables

File: test/officerManager.js (hunk 4 above)
Comment (ThaiQ) [reply test/officerManager.js]: no token require for submit

File: test/officerManager.js (hunk 5 above)
Comment (ThaiQ) [reply test/officerManager.js]: changed to 400

File: test/officerManager.js (hunk 6 above)
Comment (ThaiQ) [reply test/officerManager.js]: no token is needed for get request

File: test/officerManager.js [hunk 10]
```diff
@@ -0,0 +1,455 @@
...
+          expect(res).to.have.status(401)
+
+          done()
+        })
+        .catch(err => {
+          throw err
+        })
+    })
+
+    it('Should return an object of all forms', done => {
```
Comment (evanugarte) [inline test/officerManager.js]: i would also add to the test case that the database returns the object you previously added. see [here](https://github.com/SCE-Development/Core-v4/blob/7eb49330115efe0619c97377f7b69382cca2d41d/test/Event.js#L148)

File: src/Pages/OfficerDB/App.js [hunk 11]
```diff
@@ -0,0 +1,85 @@
+import React from 'react'
+import './App.css'
+import 'bootstrap/dist/css/bootstrap.min.css'
+import { Table } from 'reactstrap'
+
+export default class App extends React.Component {
```
Comment (evanugarte) [inline src/Pages/OfficerDB/App.js]: Please also write test cases for this component. You can use [this example](https://github.com/SCE-Development/Core-v4/blob/7eb49330115efe0619c97377f7b69382cca2d41d/test/frontend/Slideshow.test.js#L13) for reference.

File: test/frontend/OfficerDB.test.js [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+import Enzyme, { mount } from 'enzyme'
+import { expect } from 'chai'
+
+import OfficerManager from '../../src/Pages/OfficerDB/App.js'
+import Adapter from 'enzyme-adapter-react-16'
+import { Table } from 'reactstrap'
+
+Enzyme.configure({ adapter: new Adapter() })
+
+describe('Officer Manager: <Table />', () => {
```
Comment (evanugarte) [inline test/frontend/OfficerDB.test.js]: please keep the same [naming convention](https://github.com/SCE-Development/Core-v4/blob/7eb49330115efe0619c97377f7b69382cca2d41d/test/frontend/Slideshow.test.js#L13) for React tests.
```suggestion
describe('<OfficerManager />', () => {
```

File: test/frontend/OfficerDB.test.js [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+  it('Should render a <Table /> component with one child', () => {
+    const wrapper = mount(<OfficerManager />)
+    expect(wrapper.find(Table)).to.have.lengthOf(1)
+  })
+
+  it('Should render a <tbody /> component with 2 child', () => {
```
Comment (evanugarte) [inline test/frontend/OfficerDB.test.js]:
```suggestion
  it('Should render a <tbody /> component with 2 children', () => {
```

**Pull request merged**